### ISSUE TITLE
Backups: less frequent?

### ISSUE THREAD

**Issue by joeflack4** [jhu-bids/TermHub#816]
## Overview
On 6/23/2024 I lowered number of days of backups stored from 35 to 14.

I think when TermHub development stops, we should lower this to a lower number. It looks like the minimum is 7.

Additionally, it'd be nice to back up like 1x/week rather than 1x/day, but it doesn't look like there's an option for that.

Also, it is set to 'geo-redundant'. I think if we didn't have it this way, it might be cheaper. But it looks like this setting for backups is coupled to the actual DB deployment, which is set to geo-redundant. Maybe that's a good thing though. And IDK how much money we'd save by changing this.

I did get set up with someone who was supposed to look into these options with me, especially the geo-redundancy (though now I understand more), but I accidentally just archived the email and now I can't find it.

## Additional information
- [Manage backups on Azure](https://portal.azure.com/#@live.johnshopkins.edu/resource/subscriptions/fe24df19-d251-4821-9a6f-f037c93d7e47/resourceGroups/JH-POSTGRES-RG/providers/Microsoft.DBforPostgreSQL/flexibleServers/termhub/backupAndRestore)

**Comment by joeflack4** [comment]
Today I also deleted some really old backups we had from like ~1+ year ago [on the BIDS OneDrive](https://livejohnshopkins.sharepoint.com/sites/BiomedicalInformaticsandDataScience/Shared%20Documents/Forms/AllItems.aspx?csf=1&web=1&e=CpDLHf&cid=3085d2d3%2Dd167%2D4521%2D93f8%2D191edcd4a14c&FolderCTID=0x012000B0EE2A898DD4184C95198454FB686495&id=%2Fsites%2FBiomedicalInformaticsandDataScience%2FShared%20Documents%2FResearch%2FProjects&viewid=51daccc9%2D8479%2D4ef4%2Da7bf%2D65b689881f3a).

**Comment by joeflack4** [comment]
How much does it cost?
https://azure.microsoft.com/en-us/pricing/details/backup/

**[My napkin math for backup](https://azure.microsoft.com/en-us/pricing/details/backup/)**:
Instance cost per 250 GB | $7.50 + storage consumed
GRS standard: 
$0.0448 per GB

our DB is 256g and we're 80% full = ~205g. We have 14 days storage, so = 2870. / 250/instance = ~12 instances. * 7.5/instance = $90
Then per gig (0.0448 * 2870= $128). 

But I don't see this or any cost for backup cost appearing anywhere.

**[DB costs when choosing size:](https://portal.azure.com/#@live.johnshopkins.edu/resource/subscriptions/fe24df19-d251-4821-9a6f-f037c93d7e47/resourceGroups/JH-POSTGRES-RG/providers/Microsoft.DBforPostgreSQL/flexibleServers/termhub/computeAndStorage)**
Compute Sku
USD 129.94/month
Standard_D2ds_v4 (2 vCores, USD 64.97 per vCore)
2 x 64.97
Storage
USD 29.44/month
Storage selected 256 GiB (USD 0.12 per GiB)
256 x 0.12
USD 159.38/month

**Monthly billing page for termhub subscription says db costs**:
[$154](https://portal.azure.com/#@live.johnshopkins.edu/resource/subscriptions/fe24df19-d251-4821-9a6f-f037c93d7e47/overview)

**Comment by joeflack4** [comment]
Recently found out that this article is correct: https://azure.microsoft.com/en-us/pricing/details/backup/

I just wasn't seeing the cost anywhere when modifying Postgres settings. It's really lame, but you can't get a good forecast of these costs without doing the manual calculations in this article. You can only accurately see the costs post-facto.

It turns out that right now with our current 512GB setup, it's about $8.50 per rolling day. I've set it to 7 days right now, so that equates to $60/mo. If we size down to 256GB (by removing our own custom backup), we can either go with a cheaper price, or increase the number of days. Or we can just get rid of these redundant backups altogether.

**Comment by joeflack4** [comment]
Closing. Here's the conclusion:

We wanted to do 2 things:
1. Remove the rolling backups from 7 days to 0.
2. Remove our redundant pre-uploaded backup from the DB in order to downsize it and save $20/mo.

For (1), it's a non-issue because you get 7 days for free and you can't change it.

For (2), we opted not to do this because if we do, it entails remaking the DB, which might require a manual migration, and also it would entail a new IP address, so we'd have to do a new setup on our local machines and change the GitHub `ENV_FILE`, and also redeploy the app. This barely, if at all, makes this change worth it. And that is assuming nothing goes wrong in the process.

So basically there are no changes needed now.
